gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy

```diff
@@ -212,6 +197,53 @@ class CreateManifestTask extends DefaultTask {
         ])))
     }
 
+    @Nullable
+    RecommendedProductDependencies readProductDepsFromManifest(String coord, File file) {
+        def manifest
+        try {
+            def zf = new ZipFile(file)
+            def manifestEntry = zf.getEntry("META-INF/MANIFEST.MF")
+            if (manifestEntry == null) {
+                logger.debug("Manifest file does not exist in jar for '{}'", coord)
+                return
+            }
+            manifest = new Manifest(zf.getInputStream(manifestEntry))
+        } catch (IOException e) {
+            logger.warn("IOException encountered when processing artifact '{}', file '{}'", coord, file, e)
+            return
+        }
+
+        def pdeps = manifest.getMainAttributes().getValue(SLS_RECOMMENDED_PRODUCT_DEPS_KEY)
+
+        if (pdeps == null) {
+            logger.debug("No pdeps found in manifest for artifact '{}', file '{}'", coord, file)
+            return null
+        }
+
+        return jsonMapper.readValue(pdeps, RecommendedProductDependencies)
+    }
+
+    @Nullable
+    RecommendedProductDependencies readProductDepsFromPdepFile(String coord, File file) {
+        try {
+            def zf = new ZipFile(file)
+            def entry = zf.getEntry(PDEPS_FILE_PATH)
+            if (entry == null) {
+                logger.debug("Pdeps file {} does not exist in jar for '{}'", PDEPS_FILE_PATH, coord)
```

Comment:
what does "pdeps" stand for?

Reply:
and where is it specified?

Reply:
product dependencies. Renamed the file & log lines to just say that, so it's obvious

Reply:
cool

Reply:
out of curiosity, why is a separate file preferred over the MANIFEST one?

Reply:
It seemed cleaner to have a bespoke file that the conjure tooling would know how to generate.